### ISSUE TITLE
Status of a cluster is not easy to find

### ISSUE THREAD

**Issue by ashoksekar07** [kubernetes-sigs/cluster-api#820]
/kind feature

**Describe the solution you'd like**
When I created a cluster using Cluster API and the AWS provider, there was no easy way to find whether my cluster was successfully created.

For example, when you create a PVC object in Kubernetes it has a state which says 'unbound' initially and then moves on to 'bound' when it is successful. Similarly, if there is a cluster level status which says 'creating' or 'pending' and then moves to 'created' or 'running', it will be easier to know whether the cluster was successfully created. I do see that there is a Provider Status which specifies the status of individual components like VPC, subnet, etc. in the provider. But a cluster level status would be easier to understand.

I think something similar to this for the machine objects will also be useful.

**Comment by jichenjc** [comment]
same issue here. I created a openstack cluster and don't know what's the status
I can check machine part first

**Comment by liztio** [comment]
@jichenjc is this completed?

**Comment by jichenjc** [comment]
yes, it's done

**Comment by jichenjc** [comment]
I forgot to add fixes keyword and I can't close this ,maybe leave to @ashoksekar07 for close ..

**Comment by ashoksekar07** [comment]
The fix seems to address the status of a machine, but not the status of a cluster. Can a similar status field be added to the cluster?

**Comment by timothysc** [comment]
TL;DR - we need a well defined state-machine for the cluster.

**Comment by chuckha** [comment]
It would really help out testing efforts to know when a cluster hits the ready state.

Right now I go into child cluster and query Nodes directly, but if I could just stay on the management cluster that would be awesome.

**Comment by ncdc** [comment]
In v1alpha2, we have cluster.status.controlPlaneInitialized that is set to true once at least 1 control plane node is up. Are there more specific things we need to do as well?

**Comment by detiber** [comment]
Ideally we'd be able to get summarized data about worker Nodes that are managed by the cluster as well

**Comment by ncdc** [comment]
@detiber do you have a thought about the format of that status?

**Comment by ncdc** [comment]
Actually @ashoksekar07 told me he's satisfied with what is implemented. Going to close. We can open a new issue for **specific** changes such as worker node summary status.

/close
